OHRM5X-2609: Fix general defects

devishke-orange opened the pull request:
Fixes for 
- OHRM5X-2610
- OHRM5X-2611
- ORHM5X-2612
- OHRM5X-2613 (Partially complete)

devishke-orange committed:
OHRM5X-2610 fix

- validate emails in ResetPasswordService.php
- sanitize user names in ResetPasswordService.php
- add new page for email failures
- route to new page in RequestResetPasswordController.php

devishke-orange committed:
OHRM5X-2611 fix

- get resetCode as a hidden input
- use existing validation of reset code to check for modified usernames

devishke-orange committed:
OHRM5X-2612 fix

- add isEntityAccessible check into attachment controller

devishke-orange committed:
OHRM5X-2613 fix

- add user active check to auth subscriber and logout if not

RajithaKumara commented on installer/Util/AppSetupUtility.php:
```diff
@@ -72,6 +73,7 @@ class AppSetupUtility
         '5.6.1' => \OrangeHRM\Installer\Migration\V5_6_1\Migration::class,
         '5.7' => \OrangeHRM\Installer\Migration\V5_7_0\Migration::class,
         '5.7.1' => \OrangeHRM\Installer\Migration\V5_7_1\Migration::class,
+        '5.8' => \OrangeHRM\Installer\Migration\V5_8_0\Migration::class,
```
Can you move 5.7.1 migration also to 5.8. Initially I decided to make 5.7.1 release and I added few fixes there, but then thought to make the release as 5.8.

devishke-orange replied:
done

RajithaKumara commented on src/plugins/orangehrmAuthenticationPlugin/Subscriber/AuthenticationSubscriber.php:
```diff
@@ -143,4 +155,35 @@ private function getControllerInstance(ControllerEvent $event)
+            return false;
+        }
+
+        return $user->getStatus();
```
Do we need to check whether the bound employee is terminated too? or is it handling somewhere else?

devishke-orange replied:
added check for terminated employee as well

devishke-orange committed:
OHRM5X-2613 fix

- add checks for terminated employee

devishke-orange committed:
move 5.7.1 migrations to 5.8

devishke-orange committed:
Update ControllerTrait.php forward to use Symfony Response type to prevent conflict with new logout logic

devishke-orange committed:
OHRM5X-2613 fix

- store auth user last modified as session variable
- log out user if session last modified differs from db last modified
- update UpdatePasswordAPI to set last modified
- route on update password screen to trigger session expiry
- add "Session Expiry" lang string

devishke-orange committed:
OHRM5X-2610 fix

- add work email validation to FE and API

devishke-orange committed:
OHRM5X-2610 fix

- validate email in email service

devishke-orange committed:
update failing tests
